Front-end/store/src/app/battle-page/battle-page.component.ts

```diff
@@ -110,6 +111,10 @@ export class BattlePageComponent implements OnInit, OnDestroy {
     }
   }
 
+  public getWinnerCost(currentCost: number, discount: number): number {
+    return Math.round(currentCost * (1 - discount / 100));
```

Comment:
I have an idea. Can you map server battle book dto to your custom battle book dto for battle book card? So we can move this info as WinnerCost, IsWinner to mapper instead calculate that in component.